pydoctor/epydoc2stan.py

```diff
@@ -314,15 +314,16 @@ def __init__(self, obj, annotations):
     @classmethod
     def from_ast_annotations(cls, obj: model.Documentable, annotations: Mapping[str, ast.expr]) -> "FieldHandler":
         linker = _EpydocLinker(obj)
-        annotations = {name: AnnotationDocstring(value).to_stan(linker)
-                       for name, value in annotations.items()}
-        return_type = FieldDesc()
-        try:
-            return_type.body = annotations.pop("return")
-        except KeyError:
-            pass
-        ret_value = cls(obj, annotations)
-        ret_value.handle_returntype(return_type)
+        formatted_annotations = {
+            name: AnnotationDocstring(value).to_stan(linker)
+            for name, value in annotations.items()
+            }
+        ret_type = formatted_annotations.pop('return', None)
+        ret_value = cls(obj, formatted_annotations)
+        if ret_type is not None:
```

Comment:
Instead of doing a `None` sentinel, couldn't we do it with exceptions?

```
try:
    ret_type = formatted_annotations.pop("return")
except KeyError:
    pass
else:
        return_type = FieldDesc()
        return_type.body = ret_type
        ret_value.handle_returntype(return_type)
```

Reply:
Since `ret_value` is needed whether the `return` annotation exists or not, its creation must happen on both flows. But if it is moved before the `pop()`, then the `return` annotation will be mixed into the parameter annotations, which isn't how `FieldHandler` is designed. So we'd either have to duplicate the `ret_value` initialization or put `ret_type = None` in the `except:` clause and keep the `if ret_type is not None:`, neither of which is appealing.

Actually `ret_value` is a poorly chosen name here: it's the return value of the method, but has nothing to do with the `return` statement, unlike `ret_type`. I'll rename it to `handler`.